src/pages/Login/index.tsx

```diff
@@ -9,61 +9,41 @@ import starsLady from '../../media/standing-lady.svg'
 import Loader from '../../components/Loader'
 import Message from '../../components/Message'
 import FormContainer from '../../components/FormContainer'
-import Role from '../../components/Role'
 import CustomButton from '../../components/CustomButton'
 import Footer from '../../components/Footer'
-import { loginEmployerRequest } from '../../redux/actions/employer'
-import { loginJobseekerRequest } from '../../redux/actions/jobseeker'
+import { loginUserRequest } from '../../redux/actions/user'
 import { AppState } from '../../redux/types'
 
 const Login = () => {
   const [email, setEmail] = useState('')
   const [password, setPassword] = useState('')
-  const [role, setRole] = useState({})
 
   const dispatch = useDispatch()
   const history = useHistory()
 
-  const employer = useSelector((state: AppState) => state.employer)
-  const jobseeker = useSelector((state: AppState) => state.jobseeker)
-  const jobseekerError = useSelector((state: AppState) => state.jobseeker.error)
-  const jobseekerLoader = useSelector(
-    (state: AppState) => state.jobseeker.loading
-  )
-  const employerError = useSelector((state: AppState) => state.employer.error)
-  const employerLoader = useSelector(
-    (state: AppState) => state.employer.loading
-  )
-
-  const handleRole = (event: React.FormEvent) => {
-    event.preventDefault()
-    if (document.getElementById('jobseeker') === event.target) {
-      setRole(jobseeker)
-    }
-    if (document.getElementById('employer') === event.target) {
-      setRole(employer)
-    }
-  }
+  const user = useSelector((state: AppState) => state.user)
+  const { error, loading } = user
+  const role = useSelector((state: AppState) => state.user.userInfo.role)
 
   const submitHandler = (e: React.FormEvent) => {
+    console.log('role1', role)
     e.preventDefault()
-    if (role === employer) {
-      dispatch(loginEmployerRequest(email, password))
-      history.push('/company/profile')
-    } else if (role === jobseeker) {
-      dispatch(loginJobseekerRequest(email, password))
+    dispatch(loginUserRequest(email, password))
+    if (role === 'jobseeker') {
+      console.log('role', role)
       history.push('/jobseeker/profile')
+    } else if (role === 'employer') {
+      console.log('role', role)
+      history.push('/company/profile')
```

Comment:
Ok, thanks. Do I put /login/local as the initial state in useState?

Reply:
Can you clarify, why do you need `login/local` state?

Reply:
By history you mean useHistory() and pass it into the action?

With this code I get error: 'Cannot find name success' and 'const role: (state: AppState) => string
This condition will always return 'false' since the types '(state: AppState) => string' and 'string' have no overlap'

function* loginUserSaga(action: LoginUserRequestAction) {
  try {
    const email = action.payload.credential.email
    const password = action.payload.credential.password
    const history = action.payload.credential.history
    const role = (state: AppState) => state.user.userInfo.role
    const res = yield axios.post('/login/local', { email, password, history })
    console.log('res', res)
    if (res.status === 200) {
      yield put(success(res.data))
      if (role === 'employer') {
        history.push('/company/profile')
      } else if (role === 'jobseeker') {
        history.push('/jobseeker/profile')
      }
    }

    yield put(loginUserSuccess(res.data.payload))
    yield LocalStorage.saveToken(res.data.payload.token)
  } catch (error) {
    yield put(loginUserFail())
  }
}

Reply:
You can not extract role from reducer this way: `const role = (state: AppState) => state.user.userInfo.role` and you don't need to. When I said role checking inside saga, I meant you can check the role that come from the response.
```
function* loginUserSaga(action: LoginUserRequestAction) {
try {
const email = action.payload.credential.email
const password = action.payload.credential.password
// you can shape the payload of loginRequest: { credential: {email and password here}, history: history object here }
const history = action.payload.history
// No need to pass history to the request
const res = yield axios.post('/login/local', { email, password })

if (res.status === 200) {
yield put(success(res.data))

  if (response.data.role === 'employer') {
     history.push('/company/profile')
     } else if (response.data.role === 'jobseeker') {
       history.push('/jobseeker/profile')
      }
}
}
```

Reply:
I have this now, but it fails to send the data to server:
function* loginUserSaga(action: LoginUserRequestAction) {
  try {
    const email = action.payload.credential.email
    const password = action.payload.credential.password
    const history = action.payload.history
    const res = yield axios.post('/login/local', { email, password })
    
    console.log('res', res)
    if (res.status === 200) {
      //@ts-ignore
      yield put(success(res.data))
      if (res.data.role === 'employer') {
        history.push('/company/profile')
      } else if (res.data.role === 'jobseeker') {
        history.push('/jobseeker/profile')
      }
    }

    yield put(loginUserSuccess(res.data.payload))
    yield LocalStorage.saveToken(res.data.payload.token)
  } catch (error) {
    yield put(loginUserFail())
  }
}